bloom/commands/git/import_upstream.py

```diff
@@ -205,6 +206,12 @@ def handle_tree(tree, directory, root_path, version):
                 file_data = show(BLOOM_CONFIG_BRANCH, os.path.join(root_path, rel_path))
             # Write file
             with open(rel_path, 'wb') as f:
+                # Python 2 will treat this as an ascii string but
+                # Python 3 will not re-decode a utf-8 string.
+                try:
+                    file_data = file_data.encode('utf-8')
+                except UnicodeDecodeError:
+                    file_data = file_data.decode('utf-8').encode('utf-8')
```

Comment:
I'm happy replacing this with an explicit python version check. I'd noticed other places, particularly in imports where exception catching was used for 2/3 compatibility and was trying to keep consistent where I could, but I like the explicit version checks better.

The explicit decode before re-encoding was the most plain way I could find to make file_data into a utf-8 string without changing the behavior of execute_command.